docs/docs/generating-nlu-data.mdx

```diff
@@ -3,3 +3,125 @@ id: generating-nlu-data
+
+
+## Conversation-Driven Development for NLU
+
+Conversation-Driven Development (CDD) means letting real user conversations guide your development. For building a great NLU model, this means:
+
+
+### Gather real data (vs. generating synthetic data)
+When it comes to building out NLU training data, developers often rely on text generation tools to quickly increase the number of training examples. While this approach can be time-saving, it makes NLU models prone to overfit data that is not representative of things a real user would say.
+
+
+To avoid such a problem, it is always a good idea to collect as much real user data as possible to use as training data. Even though your bot will make mistakes initially, this process of training & evaluation on user data will enable your model to generalize much more effectively in real-world scenarios.
+
+### Share with test users early (vs. only trying it out yourself)
+
+In order to gather real data, you’re going to need real user messages. This means sharing your bot with test users outside the development team early on. See [these guidelines](./cdd) for more details.
+
+
+## Avoiding Intent Confusion
+
+Intents should be designed to be as distinct as possible so that the model is able to make predictions correctly
+
+
+Entities can be used to perform conditional logic for a given intent
+
+
+A common mistake developers make is creating separate intents when entities can be used as the differentiating factor.
+An example of this is when designing intents centered around the user providing information. A simple solution would be to create separate intents depending on what information is provided, e.g. inform_name, inform_address, inform_email, when all of these intents should actually be grouped as a single inform intent with several possible entity values.
+
+
+Extracted entity values can be used to affect the conversation flow using categorical slots:
+Code snippet:
+
+```story
+- inform
+    - slot{pet_type: "dog"}
+    - utter_dogs_are_cool
+
+- inform
+   - slot{pet_type: "cat"}
+   - utter_cats_are_cool_too
+```
+
+## Extracting Entities
+
+### Using Synonyms Wisely
+
+Adding synonyms to the training data is useful for mapping certain entity values to a single normalized entity
+
+Synonyms, however, are not meant for improving your model's entity recognition and has no effect on NLU performance
+
+
+A good use case for synonyms is when normalizing entities belonging to distinct groups. For example, in an assistant that asks users what insurance policies they're interested in, they might respond with "my truck," "a car," or "I drive a batmobile." It would be a good idea to map truck, car, and batmobile to the normalized value auto so that the processing logic will only need to account for a narrow set of possibilities (see [Entity Synonyms](./training-data-format/#entity-synonyms)).
```

Comment:
```suggestion
A good use case for synonyms is when normalizing entities belonging to distinct groups. For example, in an assistant that asks users what insurance policies they're interested in, they might respond with "my truck," "a car," or "I drive a batmobile." It would be a good idea to map `truck`, `car`, and `batmobile` to the normalized value `auto` so that the processing logic will only need to account for a narrow set of possibilities (see [Entity Synonyms](./training-data-format/#entity-synonyms)).
```